src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts

```diff
@@ -554,11 +558,74 @@ export function useSwapInputsController({
   };
 
   const onExecuteSwap = async () => {
-    if (!assetToSell.value || !assetToBuy.value || !quote.value) return;
+    const quoteData = quote.value as Quote | CrosschainQuote;
+    /**
+     * NOTE: Before executing a swap, we want to check several things:
+     * 1. assetToSell && assetToBuy addresses match the quote values
+     * 2. outputNative matches quote value
```

Comment:
this check makes sense but not sure if practical, like isnt there a case where the quote could update

Reply:
outputNative is updated when the quote updates

Reply:
Probably not a necessary check, but is more of a sanity check that the user is displayed and can confirm what will actually get received.